### DISCUSSION TITLE
update value to useState and got undefined!

### DISCUSSION THREAD

**Discussion by NarithChesda** [smartcontractkit/full-blockchain-solidity-course-py#1390] [Q&A]
the result print out in console, but when I update the value using useState update function, it set undefined.

```TS
export type Advertise = {
  activeBuyOrder: boolean
  advertisingBalance: BigNumber
  advertisingStatus: boolean
  appealing: boolean
  buyOrderAmountRequested: BigNumber
  buyer: string
  feeDeposited: BigNumber
  id: number
  max: BigNumber
  min: BigNumber
  rate: string
  token: string
  user: string
}

export const AdvertiseTable = ({ contract }: any) => {
  const [loading, setLoading] = React.useState(true)
  const [advertisements, setAdvertisements] = useState<Advertise[]>()

  const loadAdvertisement = async () => {
    let results = await contract.viewAdvertisements()
    setAdvertisements(results)
    console.log(results)
    console.log(advertisements)
    setLoading(false)
  }

```

**Comment by cromewar** [comment]
Hello @Kuchy300 
Changing state of booleans in react could get tricky.
Does this help you?
https://stackoverflow.com/questions/67544548/react-setstate-doesnt-update-boolean-variable

**Reply by NarithChesda** [reply]
No, the undefined line is setAdvertisement, console.log(result) print out objects, console.log(advertisements) is undefined.

**Reply by cromewar** [reply]
Oh I see, have you tried to make that call asynchronous and wait for result ?

**Reply by NarithChesda** [reply]
<img width="344" alt="Screenshot 2022-04-12 010950" src="https://user-images.githubusercontent.com/90504714/162802609-7ef6b95f-14a1-4e73-a0eb-66e82740cd89.png">

```
var ads // Declaration here

  const loadAdvertisement = async () => {
    let results = await contract.viewAdvertisements()

    console.log(results)
    console.log(results.length)

    ads = results // assign value here

    console.log(ads) //call here, it has value

    setLoading(false)
  }

  console.log(ads) // call here is undefined
```
I'm very new to react and javascripts, there might be something I don't understand. can you tell me why is this happening?

**Reply by NarithChesda** [reply]
that viewAdvertisements() returns (Advertise[ ] memory), what should be the appropriate syntax for useState declaration?  I tried 
```
const [advertisements, setAdvertisements] = useState<Advertise[]>()
```
